addOns/custompayloads/src/main/java/org/zaproxy/zap/extension/custompayloads/CustomPayloadsMultipleOptionsTablePanel.java

```diff
@@ -103,6 +120,47 @@ public void actionPerformed(ActionEvent evt) {
         addButton(resetButton);
     }
 
+    private void addPayloadFileButton() {
+        fileButton = new JButton(ADD_MULTIPLE_PAYLOADS_BUTTON);
+        fileButton.addActionListener(
+                actListnr -> {
+                    CustomPayload multiplePayloads = new CustomPayload(-1, true, "", "");
+                    Pair<File, Boolean> dialogPair = showMultiplePayloadDialog(multiplePayloads);
+                    File file = dialogPair.first;
+                    boolean preventDuplicates = dialogPair.second;
+                    if (file == null) {
+                        return;
+                    }
+                    try (BufferedReader txtReader = new BufferedReader(new FileReader(file))) {
+                        String line;
+                        while ((line = txtReader.readLine()) != null) {
+                            CustomPayload newPayload =
+                                    new CustomPayload(multiplePayloads.getCategory(), "");
+                            newPayload.setPayload(line);
+                            tableModel.setNextIdToPayload(newPayload);
+                            tableModel.addToTable(newPayload, preventDuplicates);
+                        }
+                    } catch (IOException e) {
+                        LOG.warn(e.getMessage(), e);
```

Comment:
The user should be informed that an error occurred.

Reply:
@thc202 do you mean like a warn dialog?

Reply:
Yes, since this is GUI functionality it's expected that the user is shown one (IMO).

Reply:
@bettercalln1ck you can do something similar to: https://github.com/zaproxy/zap-extensions/blob/30502523679baea6464c799dfe58b358e245701c/addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java#L163-L171